PublicFolders/src/SourceSideValidations/Get-ItemCountJob.ps1

```diff
@@ -0,0 +1,93 @@
+﻿# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+function Get-ItemCountJob {
+    [CmdletBinding()]
+    param (
+        [Parameter(Position = 0)]
+        [string]
+        $Server,
+
+        [Parameter(Position = 1)]
+        [string]
+        $Mailbox,
+
+        [Parameter(Position = 2)]
+        [PSCustomObject[]]
+        $Folders
+    )
+
+    begin {
+        $retryDelay = [TimeSpan]::FromMinutes(5)
+        $WarningPreference = "SilentlyContinue"
+        Import-PSSession (New-PSSession -ConfigurationName Microsoft.Exchange -ConnectionUri "http://$Server/powershell" -Authentication Kerberos) | Out-Null
+        $startTime = Get-Date
+        $progressCount = 0
+        $sw = New-Object System.Diagnostics.Stopwatch
+        $sw.Start()
+        $progressParams = @{
+            Activity = "Getting public folder statistics"
+        }
+
+        $itemCounts = New-Object System.Collections.ArrayList
+        $errors = New-Object System.Collections.ArrayList
+    }
+
+    process {
+        $ErrorActionPreference = "Stop" # So our try/catch works
+        $itemCounts = New-Object System.Collections.ArrayList
+        foreach ($folder in $Folders) {
+            $progressCount++
+            if ($sw.ElapsedMilliseconds -gt 1000) {
+                $sw.Restart()
+                Write-Progress @progressParams -Status $progressCount
+            }
+
+            $maxRetries = 5
+            for ($retryCount = 1; $retryCount -le $maxRetries; $retryCount++) {
+                try {
+                    $stats = Get-PublicFolderStatistics $folder.EntryId | Select-Object EntryId, ItemCount
+                    [void]$itemCounts.Add($stats)
+                    break
+                } catch {
+                    # Only retry Kerberos errors
+                    if ($_.ToString().Contains("Kerberos")) {
```

Comment:
Is "Kerberos" only going to be in English?

Reply:
I have no idea. This code path has only been used in one specific environment. We'll have to cross that bridge when we get another case where -SlowTraversal is needed.

Reply:
sounds good. Just figured i should bring that up as we have seen that with Rob's script.